apps/condo/domains/miniapp/access/B2BAccessToken.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Generated by `createschema miniapp.B2BAccessToken 'sessionId:Text; context:Relationship:B2BAppContext:CASCADE; rightSet:Relationship:B2BAppAccessRightSet:CASCADE; expiresAt:Text;'`
+ */
+
+const { userIsAdmin } = require('@open-condo/keystone/access')
+const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
+
+const { canReadObjectsAsB2BAppServiceUser, canManageObjectsAsB2BAppServiceUser } = require('@condo/domains/miniapp/utils/b2bAppServiceUserAccess')
+const { SERVICE } = require('@condo/domains/user/constants/common')
+
+async function canReadB2BAccessTokens (args) {
+    const { authentication: { item: user } } = args
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+    if (user.isAdmin) return true
+
+    if (user.type === SERVICE) {
+        return canReadObjectsAsB2BAppServiceUser(args)
+    }
+
+    return false
+}
+
+async function canManageB2BAccessTokens (args) {
+    const { authentication: { item: user } } = args
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+    if (user.isAdmin) return true
```

Comment:
Why support is not included? Or directly-accessed to be more precise

Reply:
We afraid to give rights for this entity for supports. Adding direct access -> adding rule in UserRightSet -> supports can give rights to themselves with no validation